### PR TITLE
MB-12621: Use DTOD for PPMs

### PR DESCRIPTION
**Pull request by pearl-truss**
### This PR: 

- Renamed GHCPlanner to HHGPlanner
- Moved zip3 equality check to HHGPlanner
- Consolidated Zip5TransitDistance and Zip3TransitDistance into just ZipTransitDistance into the new HHGPlanner 
- Removed check for zip3 equality from the `distance_zip_sit_origin_lookup.go` and `distance_zip_sit_dest_lookup.go` and called the new `ZipTransitDistance function
- Consolidated the `distance_zip3_lookup.go` and `distance_zip5_lookup.go` into the new `distance_zip_lookup.go`
- Removed the Bing planner that was no longer being used

### Rename GHC Planner to HHG Planner
- [api.go](https://github.com/transcom/mymove/pull/8699/files#diff-df0e528d0e9c41df8c421b2cab73303abd2f7861d9b405d683a89d1de484d0f2)
- [config.go](https://github.com/transcom/mymove/pull/8699/files#diff-17e54cc2ca859515e6a073d75da8708e25a6ca9e43d9aa5c092559219a1bf8f3)
- [hhg_planner.go](https://github.com/transcom/mymove/pull/8699/files#diff-2b6e89aef2c01ce37845412e6d684b7e5388a706f78cd4186d6629f99072df58)
- [planner.go](https://github.com/transcom/mymove/pull/8699/files#diff-78fdc41efd9be4815a22a3e7a82f3c437481a1195535b955016d215f3895d514)

### Create DTOD Planner
- [dtod_planner.go](https://github.com/transcom/mymove/pull/8699/files#diff-1168f3a4ca68a09a85a2339e4ba235d1396db733251ed3e866696fdb52ff477d)

### Use DTOD Planner for PPMs
- [ghcapi/api.go](https://github.com/transcom/mymove/pull/8699/files#diff-c77842e3be4ffa16860a440129b246e89c45661a279a368086141bf343ef4228)
- [internalapi/api.go](https://github.com/transcom/mymove/pull/8699/files#diff-48412d16cf3014f6cf94d986876a8ede5954df6b25a20e27ef0793e7faedf1a2)
- [primeapi/api.go](https://github.com/transcom/mymove/pull/8699/files#diff-81b8cc559d64d86c11d45dada1d63c207bf9fc5aae90b5a5680eb3e6376e3ea1)

#### Distance Lookup Changes (create and use new DistanceZipLookup)
- [distance zip lookup (NEW)](https://github.com/transcom/mymove/pull/8699/files#diff-273c985eb96393188de58e7e495a5e9863fa7cd6fa777b1bad0931224c13e7c6)
- [distance sit destination lookup](https://github.com/transcom/mymove/pull/8699/files#diff-4516903adb228514a19187a02f17c48444654704d8d1c324c402cf189087a169)
- [distance sit origin lookup](https://github.com/transcom/mymove/pull/8699/files#diff-0161cffe4144e7385f256f7903444d7917966cdd7f062a3bb9f2b309c91992b9)

#### Pricer Changes
- [linehaul](https://github.com/transcom/mymove/pull/8699/files#diff-d8204dfa9a51804dcd0f54d99c2c746f9041dff1964ba325dc550b329d208232)
- [shorthaul](https://github.com/transcom/mymove/pull/8699/files#diff-f29967cba80ff25906ff46b2b551783f9e5cae6b7950bd43542696f7cc446dbc)
- [fuel surcharge](https://github.com/transcom/mymove/pull/8699/files#diff-1c3a654f527bc82d826df66f1b6d882229e8beb1200574aa72e3d613c4c3a5ea)

#### Testdatagen Changes
- [e2ebasic](https://github.com/transcom/mymove/pull/8699/files#diff-d7e92054d5c6a0887e9cc47d2bad5f718c2da74b23a21f9f7753967ae0d76b73)
- [shared.go](https://github.com/transcom/mymove/pull/8699/files#diff-e24b52a37c80dbc5e9d03237bb69a26607a061212b3ecf3e17636d9cae57448c)
- [make_mto_service_item.go](https://github.com/transcom/mymove/pull/8699/files#diff-3dc2ffcb62194059056c660976d0d59c9d38fe2756117c0ed9ea7b0f023dff27)

#### Frontend Changes 
- [helpers.js](https://github.com/transcom/mymove/pull/8699/files#diff-6f58576ebf5749105cedb27ee128072892fb056983738abae4b42f042a358ae8)
- [serviceItemTestParams.js](https://github.com/transcom/mymove/pull/8699/files#diff-3594687a01fac984252eca4990e6f2c3b9bcddf28d1b56a29675b26f0b54f833)
- [serviceItems.js](https://github.com/transcom/mymove/pull/8699/files#diff-74c4f4cc0af72b766724634d3ba8028e4ac17760a22cb80250276353fe0fa874)

#### YAML Changes
- [serviceItemParamName.yaml](https://github.com/transcom/mymove/pull/8699/files#diff-f8c60feb6eaca40aa5a760904f481299e0ef47b19b1d55e7011c307fb599b8fa)
- [ghc.yaml](https://github.com/transcom/mymove/pull/8699/files#diff-51351200241c5e4af904bef502c6fc879f654c8f882d6f4a15e6ff32a521f770)
- [prime.yaml](https://github.com/transcom/mymove/pull/8699/files#diff-070fd5f7fc6e45169416a58ed811ac997012e5954a640763a2039ac8770bd602)

#### DB Migration
- [migration](https://github.com/transcom/mymove/pull/8699/files#diff-79d1eecc432acc86e1dfe8a211a10ec47ebdfa83faaee3a3f82429f69bc729d5)
#### Other Changes
- [models/service_item_param_keys.go](https://github.com/transcom/mymove/pull/8699/files#diff-97708b2c3383a0e9f20b2de6e77c651b831c0d0d140356f3ebe8a8052e62ba10)
- [ghc_payment_request_invoice.go](https://github.com/transcom/mymove/pull/8699/files#diff-cd7fccf038f1ebd436ba86a5c2906b2af226d010e08b551df7f16b4270595480)

### Updates to Integration Tests
In general this PR updates the existing PPM integration test to pass through the estimated weights page with data that is the same as what's on the existing shipment. This was done because ppm pricing depends on DTOD which we don't want to call from cypress since it's a 3rd party service. And this works since we don't run the calculate ppm pricing function when the data entered on the form matches what's on the existing shipment. 

- [entireShipment.js](https://github.com/transcom/mymove/pull/8699/files#diff-f43b0c1e003787c3e496feeb2c93e7d3a199b95ed66930a5c4d41c45dbc8824a)
- [estimatedWeightsProgear.js](https://github.com/transcom/mymove/pull/8699/files#diff-f8e191f486f693ec5cb238d60733e00524f03916859facfc44485ae57a30384c)
- [ppmCustomerShared.js](https://github.com/transcom/mymove/pull/8699/files#diff-6030762e8ba191937b6a5b4cd8af0a31704341cd23207af9cbb4f3ca62d2db3d)

## Setup to Run Your Code

<details>
<summary>💻 You will need to use three separate terminals to test this locally.</summary>

##### Terminal 1

Start the Storybook locally.

```sh
make storybook
```

##### Terminal 2

Start the UI locally.

```sh
make client_run
```

##### Terminal 3

Start the Go server locally.

```sh
make server_run
```

</details>

### Additional steps

1. Login as a PPM user such as dates_and_locations@ppm.unsubmitted
2. On the Date and Location page choose zip codes that have the same first 3 digits 

## Verification Steps for Reviewers
Create a PPM and verify that a PPM can be estimated for a shipment that has 
 - the same zip3s
 - differing zips

To test an HHG with the same details:

1. Login to the office app by creating a new prime simulator office user
2. Find the move with locator #MISHIP. This move has an HHG shipment without a destination address, missing dates (It has a requested pickup date of 3-15-2020), and missing weights.
3. We'll need to update the shipment with fields to match our PPM shipment so give it pickup dates of 3-15-2020 and an actual weight of 5,000 lbs and a destination address with the postal code 30813.
4. Create a new payment request for the move and select all of the shipment service items (exclude the basic service items if you want the payment request total to match and not just the summed shipment service items.
5. Logout of the prime simulator user and log in as the TIO.
6. Find the payment request for the move with the locator #MISHIP in the TIO queue and select it
7. View that the total is close to the PPM total for the locations, dates, and weight. The difference in price comes down to the HHG being priced with Rand McNally planner and the PPM using DTOD

## Price verification
NOTE: There's a slight differences in price that's expected since PPM pricing relies on DTOD but HHG prices only use DTOD when the zip 3s are the same 

### Differing Zip3s (prices are slightly difference since HHGs use the Rand McNally planner in this case)
Screen Shot 2022-06-08 at 10 30 49 AM
Screen Shot 2022-06-08 at 10 44 41 AM

### Same Zip3s (prices are the same since both HHGs and PPMs use the DTOD Planner in this case)
Screen Shot 2022-06-08 at 12 20 43 PM
Screen Shot 2022-06-08 at 12 24 18 PM

### CODE REVIEW

File: pkg/route/hhg_planner.go [hunk 1]
```diff
@@ -3,57 +3,55 @@ package route
...
 
 	panic("implement me")
 }
 
-// Zip5TransitDistance calculates the distance between two valid Zip5s
-func (p *ghcPlanner) Zip5TransitDistance(appCtx appcontext.AppContext, source string, destination string) (int, error) {
-	dtod := NewDTODZip5Distance(p.dtodUsername, p.dtodPassword, p.soapClient)
-	return dtod.DTODZip5Distance(appCtx, source, destination)
+// Zip3TransitDistance calculates the distance between two valid Zip5s; it is used by the PPM flow
+func (p *hhgPlanner) Zip3TransitDistance(appCtx appcontext.AppContext, source string, destination string) (int, error) {
```
Comment (gidjin) [inline pkg/route/hhg_planner.go:47]: Question: What work is needed to remove these old function signatures from the interface? Is it just removal of the HERE planner? Note if so that is outside the scope of this story, imo.

File: pkg/route/dtod_planner.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+	soapClient   SoapCaller
+	dtodUsername string
+	dtodPassword string
+}
+
+// TransitDistance calculates the distance between two valid addresses
+func (p *dtodPlanner) TransitDistance(appCtx appcontext.AppContext, source *models.Address, destination *models.Address) (int, error) {
+	// This might get retired after we transition over fully to GHC.
+
+	panic("implement me")
```
Comment (gidjin) [inline pkg/route/dtod_planner.go:21]: Question: Would it be clearer to also have the panic message state this might be deprecated soon?

File: cypress/integration/milmove/ppms/advances.js [hunk 3]
```diff
@@ -1,94 +0,0 @@
-import {
```
[the remark this answers is not in the document]
Comment (pearl-truss) [reply cypress/integration/milmove/ppms/advances.js:1]: Yes they were temporarily removed while I was testing the failing integration test in CI, I'll add them back before this is marked as Ready

File: src/components/Office/ServiceItemCalculations/helpers.test.js [hunk 4]
```diff
@@ -231,7 +231,7 @@ describe('makeCalculations', () => {
         ],
       },
       {
-        value: '32210',
+        value: '210',
```
[the remark this answers is not in the document]
Comment (pearl-truss) [reply src/components/Office/ServiceItemCalculations/helpers.test.js:234]: This value reflects milage and not a postal code

[the remark this answers is not in the document]
Comment (pearl-truss) [reply src/components/Office/ServiceItemCalculations/helpers.test.js]: This value reflects mileage and must match what's returned from the distanceZip lookup which was returning 210. But I don't know if that's confusing so I switched it to 32210

File: pkg/route/hhg_planner.go (hunk 1 above)
Comment (pearl-truss) [reply pkg/route/hhg_planner.go:47]: Agreed, the HERE planner would need to be removed. And I actually tried to remove the HERE planner in this PR but found that it was being used in a few of the older migrations (if I remember correctly it's being used to populate the duty locations table)

**Comment by duncan-truss** [comment]
@pearl-truss I was looking into the CI failure and it looks like here's our stumbling block https://github.com/transcom/mymove/pull/8699/files#diff-ed3cf02cee06e6039aaef51aa601d6015532e34e5465a0cbf20ab0a5df307451R131 .  Looks like I was logging the wrong error message, if you change `zap.Error(err)` to `zap.Error(valueErr)` maybe we can see the true error in the output.

https://app.circleci.com/pipelines/github/transcom/mymove/42500/workflows/c4d765e0-2f8c-4d2a-af2c-dbedbab5daf3/jobs/664514/parallel-runs/0/steps/0-104

**Comment by duncan-truss** [comment]
@pearl-truss I was looking into the CI failure and it looks like here's our stumbling block https://github.com/transcom/mymove/pull/8699/files#diff-ed3cf02cee06e6039aaef51aa601d6015532e34e5465a0cbf20ab0a5df307451R131 .  Looks like I was logging the wrong error message, if you change `zap.Error(err)` to `zap.Error(valueErr)` maybe we can see the true error in the output.

https://app.circleci.com/pipelines/github/transcom/mymove/42500/workflows/c4d765e0-2f8c-4d2a-af2c-dbedbab5daf3/jobs/664514/parallel-runs/0/steps/0-104

**Comment by pearl-truss** [comment]
> @pearl-truss I was looking into the CI failure and it looks like here's our stumbling block
> 
> https://github.com/transcom/mymove/blob/f13c98e2dd63844208024700cea0c5fd5ffff506/pkg/services/ppmshipment/ppm_estimator.go#L129-L133
> 
> Looks like I was logging the wrong error message, if you change `zap.Error(err)` to `zap.Error(valueErr)` maybe we can see the true error in the output.
> https://app.circleci.com/pipelines/github/transcom/mymove/42500/workflows/c4d765e0-2f8c-4d2a-af2c-dbedbab5daf3/jobs/664514/parallel-runs/0/steps/0-104

Thanks Duncan, I made the change. Is there a way to check logs for in CI runs? I'm only familiar with the artifacts tab https://app.circleci.com/pipelines/github/transcom/mymove/42510/workflows/5e06a924-d2b5-4a4d-b238-01bda0990668/jobs/664750/artifacts

**Comment by duncan-truss** [comment]
> > @pearl-truss I was looking into the CI failure and it looks like here's our stumbling block
> > https://github.com/transcom/mymove/blob/f13c98e2dd63844208024700cea0c5fd5ffff506/pkg/services/ppmshipment/ppm_estimator.go#L129-L133
> > 
> > Looks like I was logging the wrong error message, if you change `zap.Error(err)` to `zap.Error(valueErr)` maybe we can see the true error in the output.
> > https://app.circleci.com/pipelines/github/transcom/mymove/42500/workflows/c4d765e0-2f8c-4d2a-af2c-dbedbab5daf3/jobs/664514/parallel-runs/0/steps/0-104
> 
> Thanks Duncan, I made the change. Is there a way to check logs for in CI runs? I'm only familiar with the artifacts tab https://app.circleci.com/pipelines/github/transcom/mymove/42510/workflows/5e06a924-d2b5-4a4d-b238-01bda0990668/jobs/664750/artifacts

@pearl-truss Yea I know I linked directly to the output before, you should be able to expand the run sever step to see the server logs during the Cypress test

Screen Shot 2022-06-02 at 2 54 21 PM

**Comment by pearl-truss** [comment]
> That error message makes me think that our CircleCI env secret isn't set for:
> 
> https://github.com/transcom/mymove/blob/009425f7bf3a3ba98676334c04ce7e4e60d78135/pkg/route/planner.go#L165

hmm that could it but this same flag has been used the ghc router

**Comment by pearl-truss** [comment]
> > That error message makes me think that our CircleCI env secret isn't set for:
> > https://github.com/transcom/mymove/blob/009425f7bf3a3ba98676334c04ce7e4e60d78135/pkg/route/planner.go#L165
> 
> hmm that could it but this same flag has been used the ghc router

@gidjin are you more familiar with setting up a router for circleci? I'm not sure how to set the ci env secret if that's the issue

**Comment by pearl-truss** [comment]
> You should be able to view the environment variable keys that are defined under the mymove project settings https://app.circleci.com/settings/project/github/transcom/mymove/environment-variables. I didn't see anything in the list, similar to how we define what's needed for HERE Maps API to work in E2E.
> 
> https://github.com/transcom/mymove/blob/a5dd9a5bc9db9e77d41b58f332742e634dda5872/.circleci/config.yml#L594-L595
> 
> https://github.com/transcom/mymove/blob/a5dd9a5bc9db9e77d41b58f332742e634dda5872/docker-compose.e2e.yml#L57-L60

Okay, I just asked old A team in [slack ](https://ustcdp3.slack.com/archives/CP6PTUPQF/p1654286964232209?thread_ts=1654098479.031039&cid=CP6PTUPQF)

File: migrations/app/schema/20220526183214_consolidate_distanceZip3_and_distanceZip5_service_item_params.up.sql [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+-- Dom Linehaul
+('6e42f73c-a0f2-4680-8d08-27471c471451',(SELECT id FROM re_services WHERE code='DLH'),(SELECT id FROM service_item_param_keys where key='DistanceZip'), now(), now()),
+-- Dom Shorthaul
+('050f4f5f-fdec-4101-9db3-087ec49fe5ce',(SELECT id FROM re_services WHERE code='DSH'),(SELECT id FROM service_item_param_keys where key='DistanceZip'), now(), now()),
+-- FSC
+('1c8aa5fc-b993-49a3-b29b-7c4d4f27b4e3',(SELECT id FROM re_services WHERE code='FSC'),(SELECT id FROM service_item_param_keys where key='DistanceZip'), now(), now());
+
+DELETE FROM service_params WHERE service_item_param_key_id = '7a99efc3-df2b-401f-ae56-f293517afbde';
+DELETE FROM service_params WHERE service_item_param_key_id = '60b0d960-eb2e-4597-846b-d97720493799';
+DELETE FROM service_item_param_keys WHERE id = '7a99efc3-df2b-401f-ae56-f293517afbde';
```
Comment (duncan-truss) [inline migrations/app/schema/20220526183214_consolidate_distanceZip3_and_distanceZip5_service_item_params.up.sql:18]: Can these `service_item_param_key`s be deleted if they are referenced by existing payment requests in the `payment_service_item_params.service_item_param_key_id` column?

Comment (pearl-truss) [reply migrations/app/schema/20220526183214_consolidate_distanceZip3_and_distanceZip5_service_item_params.up.sql]: I think you're right and just pushed an update

File: cypress/integration/milmove/ppms/entireShipment.js [hunk 6]
```diff
@@ -56,13 +56,25 @@ describe('Entire PPM onboarding flow', function () {
...
+
+  viewportType3.forEach(({ viewport, isMobile, userId }) => {
+    it(`deletes shipment - ${viewport}`, () => {
+      if (isMobile) {
+        setMobileViewport();
+      }
+
+      navigateDeletingShipment(userId, isMobile);
+    });
+  });
```
Comment (ronaktruss) [inline cypress/integration/milmove/ppms/entireShipment.js:73]: Nit (non-blocking): If we are just doing a separate test for deletes, [homepage.js](https://github.com/transcom/mymove/blob/master/cypress/integration/mymove/homepage.js) might be a better place to have this test.

File: cypress/integration/milmove/ppms/entireShipment.js [hunk 7]
```diff
@@ -56,13 +56,25 @@ describe('Entire PPM onboarding flow', function () {
...
 });
 
-function navigateHappyPathWithDelete(userId, isMobile = false) {
-  cy.apiSignInAsUser(userId);
-  cy.wait('@getShipment');
-  customerDeletesExistingShipment();
-  customerStartsAddingAPPMShipment();
```
Comment (ronaktruss) [inline cypress/integration/milmove/ppms/entireShipment.js:65]: nit: We are no longer using the `customerStartsAddingAPPMShipment` that is imported.

**Comment by pearl-truss** [comment]
> This generally LGTM (haven't reviewed the changes to fix Duncan's comment above). Is there a way to validate that the estimator is giving values that are correct or approximately correct?

In the code? Probably not. All test would use a mock planner
In this pr? Yes, I'll add this to the description. The PPM price should be pretty similar to the price of an HHG that has only 6 auto generated service items (there may be a slight difference since an HHG move uses the rand mcnally planner when the zip3s for both pickup and destination are the same)

File: pkg/payment_request/service_param_value_lookups/weight_billed_lookup.go [hunk 8]
```diff
@@ -49,8 +49,11 @@ func (r WeightBilledLookup) lookup(appCtx appcontext.AppContext, keyData *Servic
 			value = applyMinimum(keyData.MTOServiceItem.ReService.Code, r.MTOShipment.ShipmentType, int(*originalWeight))
 		}
 	default:
-		// Make sure there's an actual weight since that's nullable but required for pricing
 		originalWeight = r.MTOShipment.PrimeActualWeight
+		if r.MTOShipment.ShipmentType == models.MTOShipmentTypePPM && r.MTOShipment.PPMShipment != nil {
+			originalWeight = r.MTOShipment.PPMShipment.EstimatedWeight
+		}
```
Comment (ronaktruss) [inline pkg/payment_request/service_param_value_lookups/weight_billed_lookup.go:55]: isn't this logic handled by https://github.com/transcom/mymove/blob/master/pkg/services/ppmshipment/ppm_estimator.go#L172?

Comment (pearl-truss) [reply pkg/payment_request/service_param_value_lookups/weight_billed_lookup.go]: agreed, I went down that rabbit hold and then realized it wasn't needed

**Comment by pearl-truss** [comment]
> I was able to validate that the HHG prices were similar to that of our DTOD prices when the zip3s don't match.
> 
> I am getting an error when generating payment requests on an HHG when using the same zip3, so haven't been able to validate this scenario. If you know of a scenario that works, please let me know. I don't see any code issues that would make this scenario not match.

I was able to verify it by locally removing the rule in the domestic line haul pricer that checks if distance is > 50 miles for HHGs. And since HHGs use DTOD when the zips are the same, the price for both HHG and PPM is the same

Screen Shot 2022-06-08 at 12 20 43 PM
Screen Shot 2022-06-08 at 12 24 18 PM

**Review by ronaktruss — APPROVED** [review]
This looks good to me. I'd probably wait on a second approval though since its a larger change. @duncan-truss / @gidjin would either of you be able to give this a second pass (specifically around the lookups).

**Comment by pearl-truss** [comment]
> @pearl-truss I think we should be removing that distance limit for PPMs. I had forgotten that wasn't done as part of the [previous story](https://dp3.atlassian.net/browse/MB-12416).

distance limits were removed for PPMs. My question is the < 50 miles should be removed from HHGs

**Comment by duncan-truss** [comment]
I think we need to add support for the DistanceZIP payment service item param key (while keeping the DistanceZIP3 and DistanceZIP5 for existing payment requests) so that the mileage shows up for new payment requests in the details.  Unless you added a migration to change all DistanceZIP3 and DistanceZIP5 to be DistanceZIP in the database then you could fully adopt DistanceZIP.  I tested this for the PARAMS move payment request by reseting the db and populating the devseed.

Screen Shot 2022-06-08 at 2 49 49 PM
)

**Comment by ronaktruss** [comment]
> > @pearl-truss I think we should be removing that distance limit for PPMs. I had forgotten that wasn't done as part of the [previous story](https://dp3.atlassian.net/browse/MB-12416).
> 
> distance limits were removed for PPMs. My question is the < 50 miles should be removed from HHGs

🤦Ahh that makes more sense

**Comment by pearl-truss** [comment]
> I think we need to add support for the DistanceZip payment service item param key (while keeping the DistanceZip3 and DistanceZip5 for existing payment requests) so that the mileage **348** shows up for new payment requests in the calculation details. Unless you added a migration to change all DistanceZip3 and DistanceZip5 to be DistanceZip in the database then you could fully adopt DistanceZIP. I tested this for the PARAMS move payment request by reseting the db and populating the devseed.

I have a fix for display the mileage on new prs. But what's the reasoning for keeping around distancezip3 and distancezip5 on "existing" PRs given that we don't actually have any

**Comment by duncan-truss** [comment]
> > I think we need to add support for the DistanceZip payment service item param key (while keeping the DistanceZip3 and DistanceZip5 for existing payment requests) so that the mileage **348** shows up for new payment requests in the calculation details. Unless you added a migration to change all DistanceZip3 and DistanceZip5 to be DistanceZip in the database then you could fully adopt DistanceZIP. I tested this for the PARAMS move payment request by reseting the db and populating the devseed.
> 
> I have a fix for display the mileage on new prs. But what's the reasoning for keeping around distancezip3 and distancezip5 on "existing" PRs given that we don't actually have any? Asking because I've foud that I can't update the service parm keys for DistanceZip5 and DistanceZip3 to use the new DistanceZip since that violates having duplicate foreign keys and I'm not sure how much time to spend on this.

That's definitely a fair question. I know I saw Reggie do it in a prior PR so that was coloring my expectation https://github.com/transcom/mymove/pull/8141/files#diff-ce668e3b100e72f9eb0a9336f67d7dbfa0f282cd7926e0873e6ced154523205aR127-R130

**Review by duncan-truss — APPROVED** [review]
Looks good, I'm glad you picked this up and not me haha.

**Comment by pearl-truss** [comment]
> > > I think we need to add support for the DistanceZip payment service item param key (while keeping the DistanceZip3 and DistanceZip5 for existing payment requests) so that the mileage **348** shows up for new payment requests in the calculation details. Unless you added a migration to change all DistanceZip3 and DistanceZip5 to be DistanceZip in the database then you could fully adopt DistanceZIP. I tested this for the PARAMS move payment request by reseting the db and populating the devseed.
> > 
> > 
> > I have a fix for display the mileage on new prs. But what's the reasoning for keeping around distancezip3 and distancezip5 on "existing" PRs given that we don't actually have any? Asking because I've foud that I can't update the service parm keys for DistanceZip5 and DistanceZip3 to use the new DistanceZip since that violates having duplicate foreign keys and I'm not sure how much time to spend on this.
> 
> That's definitely a fair question. I know I saw Reggie do it in a prior PR so that was coloring my expectation. I think there are benefits to being able to get rid of the old references to ZIP3 and ZIP5 and not have that hang around in how you've cleaned things up https://github.com/transcom/mymove/pull/8141/files#diff-ce668e3b100e72f9eb0a9336f67d7dbfa0f282cd7926e0873e6ced154523205aR127-R130

thanks for the link! I'll try adding that before merging this in
